contribs/gmf/src/services/permalink.js

```diff
@@ -579,7 +579,13 @@ gmf.Permalink.prototype.initLayers_ = function() {
   //     to any change happening to the existing layers and any added or
   //     removed ones.
   layers.forEach(function(layer) {
-    this.registerLayer_(layer, true);
+    if (layer instanceof ol.layer.Group) {
+      layer.getLayers().forEach(function(l) {
+        this.registerLayer_(l, true);
+      }, this);
+    } else {
+      this.registerLayer_(layer, true);
+    }
```

Comment:
Managing this upon initialization is not enough (I think).  Later on, if layers are added/removed to the 'data' layer group or even to the inner sub group layers, the handler methods (that call registerLayer_ and unregisterLayer) won't support adding/removing `ol.layer.Group` instances.   I think the management should occur at that level, i.e. in those methods.  They could be responsible of calling themselves (registerLayer_ should call registerLayer_ and so forth with unregisterLayer_) when dealing with a group.  That way, we would also support infinite number of layer levels.

The ADD and REMOVE events also need to be registered for the collections of layers of those groups (I'm pretty sure it's required to do so).